### ISSUE TITLE
Attribute Order is ignored, test fixtures (and tests) executed in alphabetic order

### ISSUE THREAD

**Issue by andrewlaser** [nunit/nunit#2521]
Environment:
all latest NUnit packages:
NuGet package NUnit.3.8.1.
NuGet package NUnit Console Runner 3.7.0 
NuGet package NUnit.Extension.NUnitProjectLoader.3.6.0
Selenium.Support.3.6.0
Selenium.WebDriver.3.6.0
NUnit3TestAdapter.3.8.0

**Runtime Environment:**
OS Version: Microsoft Windows NT 6.3.9600.0
CLR Version: 4.0.30319.42000

In the following example tests from testfixture ApiGiffGaffCases are started BEFORE SippCustomerSignUpCases

**Command line:** "C:\Program Files (x86)\NUnit.org\nunit-console\nunit3-console.exe" --teamcity Tests.dll --workers=1 --where "method=~ /.*/"

    _[TestFixture]
    [Parallelizable]
    public class ApiRetailCases : TestBase
    {
        [Test]
        public void Retail_Journey()
        .....
     }
    }

    [TestFixture]
    [Parallelizable]
    [Order(1)]
    public class SCustomerSignUpCases : TestBase
    {
        [Test]
        public void SCustomerSignUp()
        {
         ....
        }
    }_

NOTE1: also tried setting [Order(1)] attribute on test case level with the same result.

NOTE2: The same execution order is reproduced when running tests in Visual Studio 2017 from ReSharper "Unit Tests" window and from Tests -> Test Explorer

**Comment by ChrisMaddock** [comment]
If you put Order(2) on the ApiRetailCases fixture, does this work?

Off the top of my head, I can’t remember what we defined for ordering when some fixtures are unspecified. I think it may be just that...unspecified.

**Comment by andrewlaser** [comment]
@ChrisMaddock 
1) Setting Order(2) in all Fixtures didn't help. Order remains alphabetic
2) according to the doc https://github.com/nunit/docs/wiki/Order-Attribute
>    Tests with an OrderAttribute argument are started before any tests without the attribute.

**Comment by CharliePoole** [comment]
Let's go back to the original complaint: how do you know that one test started before the other? It's not something that is visible using the command-line settings you used. What do you see if you specify `--labels:Before`?

BTW, Since order is a framework function, we should debug this using the console-runner only because it interferes least with what you get from the framework.

**Comment by andrewlaser** [comment]
I know the actual tests order from the log I see in cmd window

Execution with suggested setting showed the following (sensitive data is skipped)
C:\Projects\RateSetter\Git\test-automation>C:\Projects\RateSetter\Git\test-automation\packages\NUnit.ConsoleRunner.3.7.0\tools\nunit3-console.exe C:\Projects\RateSetter\Git\test-automation\RS.Tests\bin\DebugStaging\RS.Tests.dll --labels:Before

Test Files
    C:\Projects\RateSetter\Git\test-automation\RS.Tests\bin\DebugStaging\RS.Tests.dll

=> RS.Tests.API.ApiMotorFinanceCases.API Motor Finance
  16:07:40,393 Thread#Worker#1 - INFO: ****          Start test: 'API Motor Finance
..... my custom log

=> RS.Tests.API.ApiMotorFinanceCases.MotorFinance_Manual_Register_Deregister_Hpi_Interest_Status

=> RS.Tests.API.ApiGiffGaffCases.GiffGaff_HandSet_E2E_Journey
..... my custom log

=> RS.Tests.Sipp.SippCustomerSignUpCases.SippCustomerSignUp
  16:11:26,045 Thread#Worker#1 - INFO: ****          Start test: 'SippCustomerSignUp'            *
..... my custom log

**Comment by CharliePoole** [comment]
Not sure what shows up in the command window... I assume you mean VS command window. The NUnit adapter doesn't produce output for each test, so it would depend on some logging you are doing yourself.

In the case of the console output, the default is that the labels appear at the __end__ of each test, since that's when the result is available. OTOH, the `--labels:before` option produces a label line when each test begins. The names of the classes in your last post are slightly different from the example, but I assume that the Order is not being honored.

`OrderAttribute` originally applied to methods only. A fairly recent change allowed it's use on fixtures. Someone will need to investigate if that change was insufficient to make it work correctly.

**Comment by jnm2** [comment]
@andrewlaser It would be super helpful to have a repro project; otherwise we have to lean on you as a debugging proxy. 

For example, can you remove selenium from the project and get the same out-of-order results? What happens if you remove `[Parallelizable]`? etc.

**Comment by CharliePoole** [comment]
@nunit/framework-team This has been here a long time without confirmation. I just came across an SO post that references it. I'm pretty sure this all comes out of confusion about what the order attribute does: controls the __start order__ of tests without guaranteeing any test will complete before any other one runs. We stated that limitation when we accepted the enhancement and the documentation is pretty clear about how it works. IOW, I'm suspecting that the user may be looking for Order to define a dependency, which is a feature we still have not implemented.
